### PR TITLE
Feature 11435 assign multiple limited diseases to users

### LINKED ISSUE #11435: Assign multiple limited diseases to users [5]
### Feature Description

image

Currently, users can be assigned a limited disease that will only allow them to see data related to this disease.

### Proposed Change

Allow users to be assigned multiple limited diseases, not only one.

### Added Value
In some countries, there is joint surveillance done on multiple diseases at the same time. Users tasked with this will be able to create and edit data related to all those diseases, but they don't need to have access to all other diseases that they don't work on.

### Acceptance Criteria

All changes have to be done in the web and mobile app.

- [x] The `User.limitedDisease` field has been renamed to `limitedDiseases` and its data type has been changed to a list of diseases
- [x] Existing data is properly migrated in `sormas_schema.sql` and `DatabaseHelper`
- [x] All code that checks access to data based on the user's limited disease now checks whether ANY of the limited diseases of the user applies; a possible approach would be to use `cb.like("%DISEASE%")` instead of `cb.equal`
- [x] In the user edit form in the web app, the limited disease field has been replaced with a new component:
  - Heading: "Disease restrictions"
  - Checkbox: "Restrict access to specific diseases"; automatically check this if the user has limited diseases; if checked, display the other elements described below
  - Explanation label: "Mark all diseases that the user is supposed to have access to"
  - Checkboxes for all active primary case-based diseases (i.e. all diseases that are currently in the dropdown) spread over three columns; if the `limitedDiseases` field of the user contains diseases that has been set to inactive, also add checkboxes for those; mark checkboxes based on the content of the `limitedDiseases` field
  - Saving the edit form updates the `limitedDiseases` field according to the checkbox values

### Implementation Details

In the web app database, store the limited diseases as a `text` data type, and write a new `DiseaseListConverter` similar to the `AgeGroupsConverter` used for disease configurations; in the mobile app, use the same pattern that we use for age groups as well, i.e. persist a `limitedDiseasesString` field, make the `limitedDiseases` field transient and use a conversion method to convert the String field to a list of diseases.

### PR DESCRIPTION
**Pull request by leventegal-she**
Fixes #11435

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12723

**Review by MateStrysewske — CHANGES_REQUESTED** [review]

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12723

**Review by MateStrysewske — APPROVED** [review]
Let's wait for the release before merging this PR

**Pull request merged**
